### PR TITLE
UOE-8449: Passing BidViewAbilityScore to Auction Endpoint

### PR DESCRIPTION
**Pull request by pm-aadit-patil**
- changes for sending bidviewability to s2s auction side

## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [x] Other

### CODE REVIEW

File: modules/prebidServerBidAdapter/index.js [hunk 1]
```diff
@@ -814,6 +814,19 @@ Object.assign(ORTB2.prototype, {
         if (adapter && adapter.getSpec().transformBidParams) {
           bid.params = adapter.getSpec().transformBidParams(bid.params, true, adUnit, bidRequests);
         }
+        // passing bid.bidViewability to pubmatic params, only when present
+        const addBidViewabilityDataS2S= function (bidRequests,bid){
+          for (let i = 0; i < bidRequests.length; i++) {
```
Comment (pm-azhar-mulla) [inline modules/prebidServerBidAdapter/index.js:819]: try to use forEach for better testability

File: modules/prebidServerBidAdapter/index.js [hunk 2]
```diff
@@ -814,6 +814,19 @@ Object.assign(ORTB2.prototype, {
         if (adapter && adapter.getSpec().transformBidParams) {
           bid.params = adapter.getSpec().transformBidParams(bid.params, true, adUnit, bidRequests);
         }
+        // passing bid.bidViewability to pubmatic params, only when present
+        const addBidViewabilityDataS2S= function (bidRequests,bid){
+          for (let i = 0; i < bidRequests.length; i++) {  
+            if( bidRequests[i].bidderCode == bid.bidder && bid.bidder=="pubmatic"){
```
Comment (pm-azhar-mulla) [inline modules/prebidServerBidAdapter/index.js:820]: use === instead of ==

File: modules/prebidServerBidAdapter/index.js (hunk 1 above)
Comment (pm-aadit-patil) [reply modules/prebidServerBidAdapter/index.js]: used forEach

Comment (pm-aadit-patil) [reply modules/prebidServerBidAdapter/index.js]: used forEach as per comment

File: modules/prebidServerBidAdapter/index.js (hunk 2 above)
Comment (pm-aadit-patil) [reply modules/prebidServerBidAdapter/index.js]: made respective changes

**Pull request closed without merging**